### PR TITLE
Demonstrate use of clang-format in LAMMPS

### PR DESCRIPTION
**Pull request by akohlmey**
**Summary**

This provides a .clang-format configuration file for the C++ sources in the `unittest` folder and reformats those source files accordingly. This is meant to serve as a demonstration for how a clang-format configuration could look like for the main parts of LAMMPS.  There is one significant difference: for the unittest sources we have agreed on 4 characters base indentation level instead of 2 as this improves readability and there are not many (deeply) nested loops like in the main source.

**Author(s)**

Richard Berger and Axel Kohlmeyer, Temple U

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

### CODE REVIEW

**Comment by akohlmey** [comment]
@sjplimp this pull request is meant to give you an impression of the kind of changes that would happen to sources in case we would adopt using clang-format throughout LAMMPS. @rbberger have iterated this to mostly resemble the LAMMPS coding style and plan to apply this to the `unittest` tree of source files. As noted in the PR description, we deliberately changed to a 4 character indentation instead of the default 2 characters.

Settings can be made for specific folders by placing a .clang-format file there, that will then apply to all files in this folder and its subdirectories. The files are applied incrementally. Whole files are updated most easily with "clang-format -i some_file.cpp", but there is also integration into programming editors.

There would even be ways to have this integrated into the git commit process, so that the clang-format command is run on every affected file automatically before git commit is executed. But that would always be a choice of the user, we can only provide instructions and config files.

**Comment by akohlmey** [comment]
@stanmoore1 you may want to check this out if it would be suitable for your KOKKOS refactor/rewrite, possibly after switching back from 4 char to 2 char indent levels. the more common and more deeply nested loops in the core LAMMPS code make a smaller base indentation result in less wrapping and thus more readable code, which is not a concern for the more "linear" codes in the unittest folders where the benefits of the larger indentation dominates.

**Comment by ellio167** [comment]
I'm definitely in favor of this. (just my 2¢) :)

**Comment by rbberger** [comment]
@akohlmey @junghans just added a `format-tests` target to format the unit tests folder

File: unittest/CMakeLists.txt [hunk 1]
```diff
@@ -5,3 +5,13 @@ add_subdirectory(force-styles)
 add_subdirectory(utils)
 
 add_subdirectory(formats)
+
+find_package(ClangFormat)
```
Comment (junghans) [inline unittest/CMakeLists.txt:9]: I would add a minimal version here, 7.0.0 had some major issues, so I would suggest 7.0.1.

**Pull request merged**